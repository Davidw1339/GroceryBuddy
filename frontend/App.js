import React from 'react';
import { YellowBox } from 'react-native';
import Navigator from './pages/Navigator';

export default class App extends React.Component {
  componentWillMount() {
    YellowBox.ignoreWarnings(['Require cycle:']);
  }

  render() {
    return (
      <Navigator />
    );
  }
}
