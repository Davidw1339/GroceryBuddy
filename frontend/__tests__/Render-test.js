// __tests__/Intro-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';


test('renders correctly', () => {
  jest.useFakeTimers();
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot('renders correctly');
});
