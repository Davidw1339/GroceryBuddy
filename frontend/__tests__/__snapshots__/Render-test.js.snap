// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    enabled={false}
    hitSlop={
      Object {
        "right": 50,
      }
    }
    maxDeltaY={20}
    minOffsetX={5}
    onGestureEvent={[Function]}
    onHandlerStateChange={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#eee",
          "flex": 1,
          "flexDirection": "column-reverse",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "paddingTop": 64,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            style={
              Object {
                "backgroundColor": "#fff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#f4f4f4",
                  "flex": 1,
                  "padding": 10,
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Array [
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "elevation": undefined,
                        "justifyContent": "flex-end",
                        "zIndex": undefined,
                      },
                    ],
                    undefined,
                  ]
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "elevation": undefined,
                      "justifyContent": "flex-end",
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "zIndex": undefined,
                    }
                  }
                />
                <View
                  pointerEvents="box-none"
                  style={
                    Array [
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        Object {
                          "elevation": undefined,
                          "justifyContent": "flex-end",
                          "zIndex": undefined,
                        },
                      ],
                      Object {
                        "alignItems": "flex-end",
                      },
                      Object {
                        "paddingVertical": 30,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "marginHorizontal": 30,
                          "zIndex": undefined,
                        },
                        Object {
                          "elevation": 5,
                          "shadowColor": "#000",
                          "shadowOffset": Object {
                            "height": 5,
                            "width": 0,
                          },
                          "shadowOpacity": 0.35,
                          "shadowRadius": 3,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      isTVSelectable={true}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "rgba(33, 150, 243, 1)",
                            "borderRadius": 28,
                            "height": 56,
                            "width": 56,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "borderRadius": 28,
                              "height": 56,
                              "justifyContent": "center",
                              "transform": Array [
                                Object {
                                  "scale": 1,
                                },
                                Object {
                                  "rotate": "0deg",
                                },
                              ],
                              "width": 56,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "rgba(255, 255, 255, 1)",
                                "fontSize": 24,
                                "marginTop": -4,
                              }
                            }
                          >
                            +
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <Modal
                animationType="fade"
                hardwareAccelerated={false}
                onRequestClose={[Function]}
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                transparent={true}
                visible={false}
              >
                <View
                  accessible={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(0, 0, 0, .4)",
                      "height": "100%",
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                  testID="RNE__Overlay__backdrop"
                />
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "white",
                        "borderRadius": 3,
                        "height": 150,
                        "padding": 10,
                        "shadowColor": "rgba(0, 0, 0, .3)",
                        "shadowOffset": Object {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowRadius": 4,
                        "width": 670,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "margin": 5,
                            }
                          }
                          theme={
                            Object {
                              "colors": Object {
                                "disabled": "hsl(208, 8%, 90%)",
                                "divider": "#bcbbc1",
                                "error": "#ff190c",
                                "grey0": "#393e42",
                                "grey1": "#43484d",
                                "grey2": "#5e6977",
                                "grey3": "#86939e",
                                "grey4": "#bdc6cf",
                                "grey5": "#e1e8ee",
                                "greyOutline": "#bbb",
                                "platform": Object {
                                  "android": Object {
                                    "error": "#f44336",
                                    "primary": "#2196f3",
                                    "secondary": "#9C27B0",
                                    "success": "#4caf50",
                                    "warning": "#ffeb3b",
                                  },
                                  "ios": Object {
                                    "error": "#ff3b30",
                                    "primary": "#007aff",
                                    "secondary": "#5856d6",
                                    "success": "#4cd964",
                                    "warning": "#ffcc00",
                                  },
                                },
                                "primary": "#2089dc",
                                "searchBg": "#303337",
                                "secondary": "#8F0CE8",
                                "success": "#52c41a",
                                "warning": "#faad14",
                              },
                            }
                          }
                        >
                          Name of List:
                        </Text>
                        <View
                          style={
                            Object {
                              "paddingHorizontal": 10,
                              "width": "100%",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "borderBottomWidth": 1,
                                "borderColor": "#86939e",
                                "flexDirection": "row",
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                ],
                              }
                            }
                          >
                            <TextInput
                              allowFontScaling={true}
                              onChangeText={[Function]}
                              placeholder="List Name"
                              style={
                                Object {
                                  "alignSelf": "center",
                                  "color": "black",
                                  "flex": 1,
                                  "fontSize": 18,
                                  "minHeight": 40,
                                }
                              }
                              testID="RNE__Input__text-input"
                              underlineColorAndroid="transparent"
                              value=""
                            />
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "borderRadius": 100,
                              "flex": 1,
                              "margin": 5,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#2089dc",
                                  "borderColor": "#2089dc",
                                  "borderRadius": 3,
                                  "borderWidth": 0,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "padding": 8,
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "white",
                                    "fontSize": 18,
                                    "paddingBottom": 1,
                                    "paddingTop": 2,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                CANCEL
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "borderRadius": 100,
                              "flex": 1,
                              "margin": 5,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#2089dc",
                                  "borderColor": "#2089dc",
                                  "borderRadius": 3,
                                  "borderWidth": 0,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "padding": 8,
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "white",
                                    "fontSize": 18,
                                    "paddingBottom": 1,
                                    "paddingTop": 2,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                ADD
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </Modal>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            onLayout={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderBottomColor": "#A7A7AA",
                "borderBottomWidth": 0.5,
                "height": 64,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "flexDirection": "row",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "bottom": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    accessibilityTraits="header"
                    allowFontScaling={false}
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, .9)",
                        "fontSize": 17,
                        "fontWeight": "600",
                        "marginHorizontal": 16,
                        "textAlign": "center",
                      }
                    }
                  >
                    Lists
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
